libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageFragment.kt

```diff
@@ -36,22 +39,57 @@ class PreviewImageFragment : Fragment() {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = inflater.inflate(layout.fragment_preview_image, container, false)
+    ): View? = inflater.inflate(R.layout.preview_image_fragment, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         val nonNullIntent = checkNotNull(requireActivity().intent)
         initializeViewModels(nonNullIntent)
-        initializeViews(nonNullIntent)
+        initializeViews()
     }
 
-    private fun initializeViews(nonNullIntent: Intent) {
-        errorLayout.setOnClickListener {
-            val lowResImageUrl = nonNullIntent.getStringExtra(ARG_LOW_RES_IMAGE_URL)
-            val highResImageUrl = nonNullIntent.getStringExtra(ARG_HIGH_RES_IMAGE_URL)
+    private fun initializeViews() {
+        initializeViewPager()
+    }
 
-            viewModel.onLoadIntoImageViewRetry(lowResImageUrl, highResImageUrl)
+    private fun initializeViewPager() {
+        val previewImageAdapter = PreviewImageAdapter(
+            loadIntoImageViewWithResultListener = { imageData, imageView, position ->
+                loadIntoImageViewWithResultListener(imageData, imageView, position)
+            }
+        )
+        previewImageViewPager.adapter = previewImageAdapter
+
+        val tabConfigurationStrategy = TabLayoutMediator.TabConfigurationStrategy { tab, position ->
+            if (tab.customView == null) {
+                val customView = LayoutInflater.from(context).inflate(layout.preview_image_thumbnail, null)
```

Comment:
Do we really want to pass null as root? Have you considered passing `root = tablayout` and `attachToParent = false`?

Reply:
Inflated layout’s root element wrapped a fixed size `ImageView`, so even if root element's layout params were ignored, the view would render fine. But I agree, we should avoid passing null. Fixed here:  83af0d7